source Aha: new fields - comments, endorsements, categories for ideas

Upgreydd opened the pull request:
## What
Adds new fields: endorsements (votes), comments for ideas, and categories for products

## How
CDK

## 🚨 User Impact 🚨
Possible to synchronize comments, endorsements (votes), and categories

marcosmarxm commented on airbyte-integrations/connectors/source-aha/source_aha/manifest.yaml:
```diff
@@ -10,6 +10,15 @@ definitions:
   selector_products:
     extractor:
       field_pointer: ["products"]
+  selector_idea_categories:
+    extractor:
+      field_pointer: ["idea_categories"]
+  selector_idea_votes:
+    extractor:
+      field_pointer: ["idea_endorsements"]
+  selector_idea_comments:
+    extractor:
+      field_pointer: ["idea_comments"]
```
```suggestion
  selector_products:
    extractor:
      field_pointer: ["{{ options['name'] }}"]
```

marcosmarxm commented on airbyte-integrations/connectors/source-aha/source_aha/manifest.yaml:
```diff
@@ -74,6 +100,40 @@ definitions:
     $options:
       name: "ideas"
       path: "/ideas"
+  idea_votes_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "idea_votes"
```
Better rename to `idea_endorsements`.

marcosmarxm reviewed (changes requested):
some comments.

Upgreydd commented:
> some comments.

Thank you @marcosmarxm. I did the cleanup of redundant parts.

marcosmarxm commented:
/test connector=connectors/source-aha

marcosmarxm commented:
@Upgreydd can you check failed tests?

Upgreydd commented:
> @Upgreydd can you check failed tests?

@marcosmarxm 
It's related to limited `API key`/`aha plan` to access `features` and `goals` options. Looks like airbyte has a limited key too (so do I). It's not related to my PR at all cause the same test fails in the master. The only option to test it without a good API key is to remove tests for `features` and `goals` I guess?

Upgreydd commented:
@marcosmarxm I've removed checks for `goals` and `features` because they are optional and my API key and airbyte API key doesn't cover that part - that's why it fails tests.
Rebased on the new version with `manifest.yaml` changes from master and fixed definitions and structure.

marcosmarxm commented:
/test connector=connectors/source-aha

phdmohamedali commented:
any updates on this

Upgreydd commented:
> any updates on this

My part works as expected, Issue is with base code. No idea why it's not checked while running CI

sh4sh commented:
Seeded streams in the integration testing account

sh4sh commented:
/test connector=connectors/source-aha

archangelic commented:
/test connector=connectors/source-aha
